### DISCUSSION TITLE
Warning: Prop href did not match. Server: "https://www.google-1.com" Client: "https://www.google-2.com"  behind a feature flag

### DISCUSSION THREAD

**Discussion by karmingc** [vercel/next.js#31440] [Help]
### What version of Next.js are you using?

12.0.1

### What version of Node.js are you using?

14.16.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Other platform

### Describe the Bug

Relates to https://github.com/vercel/next.js/issues/22130.

Assuming we have a feature flag and we want to update the href for a `<a>` element.

Without feature flag, href will be https://www.google-1.com
With feature flag, href will be https://www.google-2.com

If we turn the feature flag on, https://www.google-2.com will be correctly displayed and no error will be raised.

If we turn off the feature flag after turning it on, we still see https://www.google-2.com persist as href while 

```
Warning: Prop href did not match. Server: "https://www.google-1.com" Client: "https://www.google-2.com"  behind a feature flag
```
is raised.

### Expected Behavior

Without feature flag, href will be https://www.google-1.com
With feature flag, href will be https://www.google-2.com

If we turn the feature flag on, https://www.google-2.com will be correctly displayed and no error will be raised.

If we turn off the feature flag after turning it on, we should see https://www.google-1.com as href and no error raised.

### To Reproduce

```
let exampleHref = "https://www.google-1.com"

if (featureFlagIsOn) {
  exampleHref = "https://www.google-2.com"
}

<a... href=`${exampleHref}`>...</a>

```

**Comment by SaniJasi** [comment]
Hi. My links were like this -> `<Link href={`/case-studies/${post?.fields?.slug}`}>`.
In my file `[slug.js]` (for case studies) I've added for `fallback: "blocking"` in `getStaticPaths` and this helped for me

**Comment by frei-x** [comment]
Check if the initial value of your href is empty on the server side, it may be changed on the client side
